File: components/component-inflearn/src/main/java/me/study/jpa/inflearn/order/OrderItem.java
package me.study.jpa.inflearn.order;

import lombok.Getter;
import me.study.jpa.inflearn.item.Item;

import javax.persistence.*;

import static javax.persistence.FetchType.*;

@Entity
@Table(name = "ORDER_ITEM")
@Getter
public class OrderItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = LAZY)
    @JoinColumn(name = "ORDER_ID")
    private Order order;

    @ManyToOne(fetch = LAZY)
    @JoinColumn(name = "ITEM_ID")
    private Item item;

    private Integer price;

    private Integer count;
}

File: components/component-kotlin/src/main/kotlin/me/study/jpa/OrderItem.kt
package me.study.jpa

import javax.persistence.*
import javax.persistence.FetchType.*

@Entity
class OrderItem(
    @Id
    @Column(name = "ID")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private var id_: Long? = null,

    @ManyToOne(fetch = LAZY)
    @JoinColumn(name = "ORDER_ID")
    private var order: Order? = null
)
